plugins/ipam/host-local/backend/allocator/range.go

```diff
@@ -54,7 +54,7 @@ func (r *Range) Canonicalize() error {
 			return err
 		}
 		subnet := (net.IPNet)(r.Subnet)
-		if !subnet.Contains(r.Gateway) {
+		if r.Gateway.To4() != nil && !subnet.Contains(r.Gateway) {
```

Comment:
I think it might be better to remove the check entirely - there are a number of plugins that have gateways outside the subnet in IPv4 as well (e.g. Calico uses a /32 on the interface inside the container with "default via gateway" and "gateway/32 onlink").

It depends on which misconfigurations we're most likely to catch here.  If it's just typos in the gateway, then @dcbw points out that the incorrect gateway could be in subnet and still break things.

WDYT?

Reply:
Yes, IMO, the `ipam` plugin doesn't need to check `gateway` in subnet.
I will remove the check.